pkgs/os-specific/bsd/freebsd/by-name/filterSource.nix

```diff
@@ -0,0 +1,11 @@
+{ lib, pkgsBuildBuild, runCommand, writeText, source }:
+{ pname, path, extraPaths ? [] }:
+let
+  sortedPaths = lib.naturalSort ([ path ] ++ extraPaths);
```

Comment:
Does rsync wants these paths to be sorted? Or is this to avoid rebuilds?

Reply:
Genuinely unsure. Another @artemist commit: https://github.com/rhelmot/nixpkgs/commit/b4b9e450a3b74d4437ddbb5e5fe2ec5b0ef24805

Maybe it was to make it possible to binary search some issue which was coming up during filtering?

Reply:
It's not technically required, but the rsync man page says "NOTE: sorting the list of files in the --files-from input helps rsync to be more efficient, as it will avoid re-visiting the path elements that are shared between adjacent entries."

This isn't generally a problem but we do sometimes have both a child and its parent in the list. It also has the added benefit of reducing rebuilds as you've mentioned.